test/ConnectionTest3.cpp

```diff
@@ -0,0 +1,77 @@
+#include <gtest/gtest.h>
+
+#include "DBConnection3.hpp"
+#include "DBConnection3Caster.hpp"
+
+namespace {
+
+    int getMySqlConnection3Info(IDBConnection3& dbConnection) {
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        return mysql_connection.query();
+    }
+
+    std::string getMySqlConnection3AdvancedInfo(IDBConnection3& dbConnection) {
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        return mysql_connection.advancedQuery().server_version;
+    }
+
+    int getSqLiteConnection3Info(IDBConnection3& dbConnection) {
+        auto& sqlite_connection = DBConnection3Caster<SqLiteDBConnection3, IDBConnection3>::cast(dbConnection);
+        return sqlite_connection.query();
+    }
+
+    void checkMySqlConnection3Info(IDBConnection3& dbConnection, CastType& assertion, int& expectedResult) {
+        ASSERT_EQ((DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::castType()), assertion);
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        ASSERT_EQ(expectedResult, mysql_connection.query());
+    }
+
+    void checkMySqlConnection3AdvancedInfo(IDBConnection3& dbConnection, CastType& assertion, std::string& expectedResult) {
+        ASSERT_EQ((DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::castType()), assertion);
+        auto& mysql_connection = DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::cast(dbConnection);
+        ASSERT_EQ(expectedResult, mysql_connection.advancedQuery().server_version);
+    }
+
+    void checkSqLiteConnection3Info(IDBConnection3& dbConnection, CastType& assertion, int& expectedResult) {
+        ASSERT_EQ((DBConnection3Caster<SqLiteDBConnection3, IDBConnection3>::castType()), assertion);
+        auto& sqlite_connection = DBConnection3Caster<SqLiteDBConnection3, IDBConnection3>::cast(dbConnection);
+        ASSERT_EQ(expectedResult, sqlite_connection.query());
+    }
+
+    TEST(MySqlConnection3InfoTest, GetInfoStaticCast) {
+        MySqlDBConnection3 mysql_connection("Some server", 7);
+        ASSERT_EQ((DBConnection3Caster<MySqlDBConnection3, IDBConnection3>::castType()), CastType::STATIC_CAST);
+        ASSERT_EQ(7, getMySqlConnection3Info(mysql_connection));
```

Comment:
@Morheit , rewrite tests to use `check...` instead` of `get...`

Reply:
@malirod , done